lua/plenary/async_lib/work.lua

```diff
@@ -0,0 +1,91 @@
+local a = require('plenary.async_lib.async')
+local uv = vim.loop
+
+local M = {}
+
+local function work_map(tbl, func, callback)
+  local results = {}
+  local ran_callback = false
+
+  local work = uv.new_work(func, function(idx, ...)
```

Comment:
I don't think we do, but we should check and make sure we don't have to do any cleanup for work here (in case leving this function or reference around could prevent garbage colleciton).